src/atomate2/vasp/jobs/adsorption.py

```diff
@@ -0,0 +1,437 @@
+    """Maker for molecule relaxation.
+
+    Parameters
+    ----------
+    name: str
+        The name of the flow produced by the maker.
+    input_set_generator: VaspInputGenerator
+        The input set generator for the relaxation calculation.
+    """
+
+    name: str = "mol_relax_maker__"
+    input_set_generator: VaspInputGenerator = field(
+        default_factory=lambda: MPRelaxSet(
+            user_kpoints_settings=Kpoints.from_dict(
+                {
+                    "nkpoints": 0,
+                    "generation_style": "Gamma",
+                    "kpoints": [[3, 3, 3]],
+                    "usershift": [0, 0, 0],
+                    "comment": "Automatic mesh",
+                }
+            ),
+            user_incar_settings={
+                "ISIF": 2,
+                "ENCUT": 700,
+                "GGA": "RP",
+                "EDIFF": 1e-5,
+                "LDAU": False,
+                "NSW": 300,
+                "NELM": 500,
+            },
+            auto_ispin=True,
+        )
+    )
+
+
+@dataclass
+class MolStaticMaker(BaseVaspMaker):
+    """Maker for molecule static energy calculation.
+
+    Parameters
+    ----------
+    name: str
+        The name of the flow produced by the maker.
+    input_set_generator: VaspInputGenerator
+        The input set generator for the static energy calculation.
+    """
+
+    name: str = "mol_static_maker__"
+    input_set_generator: VaspInputGenerator = field(
+        default_factory=lambda: MPRelaxSet(
+            user_kpoints_settings=Kpoints.from_dict(
+                {
+                    "nkpoints": 0,
+                    "generation_style": "Gamma",
+                    "kpoints": [[3, 3, 3]],
+                    "usershift": [0, 0, 0],
+                    "comment": "Automatic mesh",
+                }
+            ),
+            user_incar_settings={
+                "ENCUT": 700,
+                "IBRION": -1,
+                "GGA": "RP",
+                "EDIFF": 1e-7,
+                "LDAU": False,
+                "NSW": 0,
+                "NELM": 500,
+            },
+            auto_ispin=True,
+        )
+    )
+
+
+@dataclass
+class SlabRelaxMaker(BaseVaspMaker):
+    """Maker for adsorption slab relaxation.
+
+    Parameters
+    ----------
+    name: str
+        The name of the flow produced by the maker.
+    input_set_generator: VaspInputGenerator
+        The input set generator for the relaxation calculation.
+    """
+
+    name: str = "slab_relax_maker__"
+    input_set_generator: VaspInputGenerator = field(
+        default_factory=lambda: MPRelaxSet(
+            user_kpoints_settings={"reciprocal_density": 200},
+            user_incar_settings={
+                "ISIF": 2,
+                "ENCUT": 700,
+                "GGA": "RP",
+                "EDIFF": 1e-5,
+                "LDAU": False,
+                "NSW": 300,
+                "NELM": 500,
+            },
+            auto_ispin=True,
```

Comment:
I'd be careful about using `auto_ispin` for this maker. If you give a prev_dir of the bulk for example, it could disable ispin even if your molecule is spin polarised (e.g., oxygen).